File: 1주차/YH/longestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        answer = ''
        strs_list = []
        
        for i in range(len(strs)) : 
            strs_list.append(strs[i])
            strs_list.sort(key=len)
            
        for i in range(len(strs_list[0])) :             
            for j in range(1, len(strs_list)) :
                if strs_list[0][i] != strs_list[j][i] : 
                    return strs[0][:i]
                
        
        return answer

File: 1주차/NH/longest-common-prefix.js
function longestCommonPrefix(strs) {
    let answer="";
    for (let i = 0; i <= strs[0].length; i++) {
        for (let j = 1; j < strs.length; j++) {
            if (strs[0][i] !== strs[j][i]) {
                strs[0]=strs[0].slice(0, i);
            }
        }
    }
    return strs[0]
};